joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/passes/DownloadDependenciesPass.scala

```diff
@@ -0,0 +1,36 @@
+package io.joern.gosrc2cpg.passes
+
+import better.files.File
+import io.joern.gosrc2cpg.model.GoModHelper
+import io.joern.gosrc2cpg.utils.AstGenRunner.getClass
+import io.joern.x2cpg.utils.ExternalCommand
+import org.slf4j.LoggerFactory
+
+import scala.util.{Failure, Success}
+
+class DownloadDependenciesPass(parentGoMod: GoModHelper) {
+  private val logger = LoggerFactory.getLogger(getClass)
+  def process(): Unit = {
+    File.usingTemporaryDirectory("go-temp-download") { tmpDir =>
+      setupDummyProjectAndDownload(tmpDir.toString)
+    }
+  }
+
+  private def setupDummyProjectAndDownload(prjDir: String): Unit = {
+    parentGoMod
+      .getModMetaData()
+      .map(mod => {
+        ExternalCommand.run("go mod init joern.io/temp", prjDir) match
+          case Success(_) =>
+            mod.dependencies.foreach(dependency => {
+              val cmd = s"go get ${dependency.module}@${dependency.version}"
+              ExternalCommand.run(cmd, prjDir) match
+                case Success(_) => print(". ")
```

Comment:
We probably want to avoid calling `print` as this cannot be configured like the logger

Reply:
When a download is happening for a little longer time and if there is no indicator to show some sort of progress on the screen. It gives the feeling of the process of getting hanged or stuck. I guess this progress bar cannot be achieved without using `print`.

Reply:
Right, fair point. I'll accept the PR as is then.